### ISSUE TITLE
[FEATURE]: Configurable folder to search for custom commands

### ISSUE THREAD

**Issue by isaacpz** [garden-io/garden#2909]
## Feature Request

All custom commands currently [must be stored in the project root](https://docs.garden.io/advanced/custom-commands#limitations). For projects with many custom commands, this makes the root folder explode with lots of files.

I understand there is a performance concern with searching all subfolders, but it'd be great if I could configure a single folder that should be used to store custom commands in. 

Ideally, subfolders would also be searched (with a much less demanding performance impact because the search path will be so restricted)
 
### Background / Motivation

I am trying to set up a Garden project that heavily uses the custom command functionality.

### What should the user be able to do?

In the project `garden.yml`, the user should be able to include something like: `customCommandPath: commands`, which is added to the search path when looking for custom commands.

For example, this folder structure should work:
```
project.garden.yml (includes customCommandPath: commands)
commands/
   command1.garden.yml
   command2.garden.yml
   command3.garden.yml
   module1/
      command4.garden.yml
command5.garden.yml
```

### Why do they want to do this? What problem does it solve?

I do not want lots of custom command configurations in the root of my project.

### Suggested Implementation(s)

See above

### How important is this feature for you/your team?

🌹 It’s a nice to have, but nice things are nice 🙂

**Comment by thsig** [comment]
Hi @isaacpz—thanks for writing this up!

@edvald, am I right in assuming that this could be done without significant performance impact?

**Comment by edvald** [comment]
Correct, this should have no performance impact since the project config is loaded before the commands are loaded.
